addons/web_editor/static/src/js/editor/transcoder.js

```diff
@@ -229,9 +246,30 @@ function classToStyle($editable) {
         } else {
             $target.attr('style', style);
         }
+        // Apple Mail
         if (this.nodeName === 'TD' && !this.childNodes.length) {
             this.innerHTML = '&nbsp;';
         }
+
+        // Outlook
+        if (this.nodeName === 'A' && $target.hasClass('btn') && !$target.children().length) {
+            var $hack = $('<table class="o_outlook_hack"><tr><td></td></tr></table>');
+            $hack.find('td')
+                .attr('height', $target.outerHeight())
+                .css('margin', $target.css('padding'))
+                .css('border-radius', $target.css('border-radius'))
+                .css('background-color', $target.css('background-color'));
+            $target.after($hack);
+            $target.appendTo($hack.find('td'));
```

Comment:
So basically you are putting the button in a table cell with the same style of the button... is not this gonna duplicate the margin for smarter mail clients ?

Reply:
The rendering is the same in the tests, both are collapsing (and one is not interpreted by some browsers)